q2_longitudinal/_longitudinal.py

```diff
@@ -184,11 +196,51 @@ def nmit(table: pd.DataFrame, metadata: qiime2.Metadata,
     metadata = metadata[metadata.index.isin(table.index)]
 
     # validate id column
-    _validate_input_columns(metadata, individual_id_column, None, None)
+    _validate_input_columns(metadata, individual_id_column, None, None, None)
 
     # run NMIT
     _dist = _nmit(
         table, metadata, individual_id_column=individual_id_column,
         corr_method=corr_method, dist_method=dist_method)
 
     return _dist
+
+
+def first_differences(metadata: qiime2.Metadata, state_column: str,
+                      individual_id_column: str, metric: str,
+                      replicate_handling: str='error',
+                      table: pd.DataFrame=None) -> pd.Series:
+
+    # find metric in metadata or derive from table and merge into metadata
+    if table is not None:
+        _validate_metadata_is_superset(metadata.to_dataframe(), table)
+        metadata = _add_metric_to_metadata(table, metadata, metric)
+    else:
+        metadata = _load_metadata(metadata)
+
+    # validate columns
+    _validate_input_columns(
+        metadata, individual_id_column, None, state_column, metric)
+
+    return _first_differences(
+        metadata, state_column, individual_id_column, metric,
+        replicate_handling, distance_matrix=None)
+
+
+def first_distances(distance_matrix: DistanceMatrix, metadata: qiime2.Metadata,
+                    state_column: str, individual_id_column: str,
+                    replicate_handling: str='error') -> pd.Series:
+
+    # load and validate metadata
+    metadata = _load_metadata(metadata)
+    _validate_metadata_is_superset(
+        metadata, _tabulate_matrix_ids(distance_matrix))
+
+    # validate columns
+    # "Distance" is not actually a metadata value, so don't validate metric!
+    _validate_input_columns(
+        metadata, individual_id_column, None, state_column, None)
+
+    return _first_differences(
+        metadata, state_column, individual_id_column, metric='Distance',
```

Comment:
Can you pass `metric=None` here? `metric` is ignored when a `distance_matrix` is provided, so passing a null value might avoid confusion about `metric` actually being used to lookup values somewhere.

Reply:
thanks for catching @jairideout !